packages/services-s3/src/mode/pure.ts

```diff
@@ -0,0 +1,69 @@
+import { ReadFile } from '@seagull/commands-s3/dist/src/read_file'
+import { IMode } from '@seagull/mode'
+import * as fs from 'fs'
+import { injectable } from 'inversify'
+import 'reflect-metadata'
+import { S3Base } from './base'
+
+/**
+ * Http pure mode implementation.
+ */
+@injectable()
+export class S3Pure extends S3Base {
+  protected mode: Readonly<IMode> = { environment: 'pure' }
+
+  /**
+   * Writes a local folder to a mocked S3 bucket
+   * @param bucketName bucket name
+   * @param filePath file path
+   * @param content file content
+   * @throws {S3Error}
+   */
+  async writeFolder(
+    bucketName: string,
+    path: string,
+    includeSubfolders = true
+  ) {
+    const pathLength = path.length
+    const fsPaths = this.listFilesFS(path, includeSubfolders)
+    for (const fsPath of fsPaths) {
+      // cut out local path
+      const s3Path = fsPath.substr(pathLength + 1)
+      const content = fs.readFileSync(fsPath, 'utf8')
+      this.writeFile(bucketName, s3Path, content)
+    }
+  }
+
+  /**
+   * Returns all files within the path.
+   * @param path path
+   * @param includeSubfolders include subfolders
+   * @param filePaths file path list (for recursion)
+   */
+  private listFilesFS(
```

Comment:
how about something like:
```
private listFilesFS(path: string, includeSubfolders = true): string[] {
    const elements = fs.readdirSync(path)
    const paths = elements.map(el => `${path}/${el}`)
    const isDirectory = (path: string) => fs.statSync(path).isDirectory()
    const files = paths.filter(path => !isDirectory(path))
    const directories = paths.filter(isDirectory)

    return includeSubfolders ? this.addDirContents(files, directories) : files
  }

  private addDirContents(files: string[], directories: string[]) {
    const contents = directories.map(dir => this.listFilesFS(dir))
    const directoryFiles = flatten(contents)
    return files.concat(directoryFiles)
  }
```

Reply:
satisfying the flat code cultists :) by splitting the function into two and replacing "if" with ternary. better readability by this in my opinion is questionable as it includes more logic, more (block) text to read and makes performance worse by using more memory and lodash functions to hide the fact this now has five loops instead of one.